### ISSUE TITLE
elfeed-search-mode-hook not defined

### ISSUE THREAD

**Issue by alphapapa** [skeeto/elfeed#247]
Hi Chris,

I see that `elfeed-search-mode-hook` is called in `elfeed-search-mode`, but it doesn't seem to be defined anywhere.

**Comment by alphapapa** [comment]
BTW, perhaps instead of (or in addition to) this hook, there could be an `elfeed-search-update-hook`.  I have a function that I use to insert overlays between entries where the date changes, and currently I run it in `elfeed-search-mode-hook`, but it doesn't run when the buffer is updated with `elfeed-search-update`, so I have to kill the search buffer and run `elfeed` again.

If you like, I could submit a PR to add one or both of these.  Thanks.

**Comment by skeeto** [comment]
It's a quaint, old Emacs convention that hook variables, particularly 
mode hooks, are not actually defined/declared anywhere (defvar, etc.). 
That's why add-hook and friends accept a symbol for the hook. These 
functions create the variable if necessary. Honestly, it would be more 
appropriate for hooks to always be plain old global variables, and for 
add-hook and remove-hook to be macros much like push and pop. That ship 
has already sailed.

There is elfeed-update-hooks, but it sounds like you definitely want 
your hook to run _after_ the search buffer has been updated. It would be 
kind of a hack to arrange that on elfeed-update-hooks, so I went ahead 
and added elfeed-search-update-hook.
